aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp

```diff
@@ -31,90 +38,103 @@ class QConvPackWeightInt8 final : public c10::OperatorKernel {
+        kSpatialDim,
+        " elements for ",
+        kSpatialDim,
+        "D convolution.");
+    TORCH_CHECK(
+        padding.size() == kSpatialDim,
+        "Specify front/top/left padding only. "
+        "end/bottom/right padding assumed to be equal to front/top/left");
+    TORCH_CHECK(
+        dilation.size() == kSpatialDim,
+        "dilation should contain ",
+        kSpatialDim,
+        " elements for ",
+        kSpatialDim,
+        "D convolution.");
+    const int output_channels = weight.size(0);
+    const int input_channels_per_group = weight.size(1);
+    const int kernel_d = kSpatialDim == 2 ? 1 : weight.size(2);
+    const int kernel_h = weight.size(kSpatialDim);
+    const int kernel_w = weight.size(kSpatialDim + 1);
 
     // mini-batch doesn't have any impact on how we pack weights
     // so we pass it as 1
     // Input image height/width also don't have any impact on how we pack
     // weights so we can pass any values
-    fbgemm::conv_param_t<2> conv_p(
-        1, // Mini-Batch
+    const fbgemm::conv_param_t<kSpatialDim> conv_p = MakeFbgemmConvParam(
         input_channels_per_group * groups, // input channels
         output_channels,
-        {28, 28}, // Image height and width
         groups,
-        {kernel_h, kernel_w},
-        {static_cast<int>(stride[0]), static_cast<int>(stride[1])},
-        {static_cast<int>(padding[0]),
-         static_cast<int>(padding[1]),
-         static_cast<int>(padding[0]),
-         static_cast<int>(padding[1])},
-        {static_cast<int>(dilation[0]), static_cast<int>(dilation[1])});
+        kSpatialDim == 2 ? std::vector<int>{kernel_h, kernel_w}
+                         : std::vector<int>{kernel_d, kernel_h, kernel_w},
+        stride,
+        padding,
+        dilation);
 
-    // FBGEMM expects weights to be in channels last
-    auto weight_contig = weight.contiguous(MemoryFormat::ChannelsLast);
     const auto qtype = weight.qscheme();
-    std::vector<int32_t> zero_points(1, 0);
+    std::vector<int32_t> zero_points;
     if (qtype == kPerTensorAffine) {
-      zero_points[0] = weight.q_zero_point();
+      zero_points = {static_cast<int32_t>(weight.q_zero_point())};
     } else if (qtype == kPerChannelAffine) {
       int64_t axis = weight.q_per_channel_axis();
       TORCH_CHECK(
           axis == 0,
           "Only per output channel quantization is supported for the weights");
-      zero_points.resize(output_channels, 0);
+      zero_points.resize(output_channels);
       for (int i = 0; i < output_channels; ++i) {
         zero_points[i] = weight.q_per_channel_zero_points()[i].item<int32_t>();
       }
     } else {
       TORCH_CHECK(false, "Unsupported qscheme: ", toString(qtype));
     }
 
-    const int8_t* weight_ptr_int8 =
-        reinterpret_cast<int8_t*>(weight_contig.data_ptr<c10::qint8>());
-
+    // FBGEMM expects weights to be in channels last
+    const Tensor weight_tensor = weight.contiguous(MemoryFormat::ChannelsLast);
```

Comment:
Will fail for 5d tensor

Reply:
Sure, I have make a fallback change here.